### DISCUSSION TITLE
Localization of module with Syncfusion components

### DISCUSSION THREAD

**Discussion by LuisVilaplana** [oqtane/oqtane.framework#3334] [General] [answered]
I managed to create a module with Syncfusion components, it's working very well, but I cannot make localization to work.

I used this code on both ServerStartups.cs and ClientStartup.cs:

        services.AddSyncfusionBlazor();
        services.Configure<RequestLocalizationOptions>(options =>
        {
            var culture = new CultureInfo("es");
            culture.NumberFormat.CurrencySymbol = "¢";
            culture.NumberFormat.NumberDecimalSeparator = ".";
            culture.NumberFormat.NumberGroupSeparator = ",";
            culture.NumberFormat.CurrencyDecimalSeparator = ".";
            culture.NumberFormat.CurrencyGroupSeparator = ",";
            options.DefaultRequestCulture = new RequestCulture(culture);
        });
        services.AddLocalization(options => options.ResourcesPath = "Resources");
        services.AddSingleton(typeof(ISyncfusionStringLocalizer), typeof(SyncfusionLocalizer));
    }

And placed the resource files in the Resources folder:

![image](https://github.com/oqtane/oqtane.framework/assets/103559699/574c2a54-8c96-4bc0-a797-f96cf6746ecc)

Have tested the same on server and client proyects, configured the debug.cmd and nuspec files, etc. but nothing seems to work

Has anyone been able to get this to work?

**Comment by sbwalker** [comment]
Localization has a lot of moving parts to consider from a deployment perspective.
 
RESX files are located in your module Client project using a specific folder convention (matching the folder/file structure of your razor components):

/Your.Module.Client
../Modules
..../Your.Module
........Index.razor
../Resources
..../Your.Module
......Index.es.resx
   
When you build the module project it will create a folder under the Client /bin which contains a satellite assembly:
 
Your.Module/Client/bin/debug/net7.0/es/Your.Module.Oqtane.resources.dll
 
The satellite assembly will need to be deployed into the Oqtane Framework Server /bin folder subfolder corresponding to the culture:
 
oqtane.framework/Oqtane.Server/bin/debug/net7.0/es/Your.Module.Oqtane.resources.dll

This can be accomplished by modifying the debug.cmd in the YourModule.Package folder to deploy the satellite assembly automatically on build– the same way it deploys the other module assemblies. Similarly you could add the DLL reference to the nuspec so that it is included in the Nuget package. 
 
When the Oqtane framework starts up it will load the module's satellite assembly (as long as the specific language has been enabled for the site in Language Management).

**Comment by LuisVilaplana** [comment]
Thanks a lot Shaun, I got it working now with your instructions, all Syncfusion static texts are in spanish, but currency symbol, thousand and decimal delimiters are wrong.

![image](https://github.com/oqtane/oqtane.framework/assets/103559699/a801283b-d21a-4c7d-a836-d6342286787c)

Any idea?

**Reply by sbwalker** [reply]
I do not have any ideas - this would require further investigation

**Comment by LuisVilaplana** [comment, marked as answer]
ok, I managed to solve it. I changed Oqtane's language to Spanish Costa Rica (es-CR). This fixed the whole problem with the currency symbol, dates and thousands and decimals separators. But obviously there was no translation of any kind, so I created a new language pack for es-CR based on the excellent work of Santiago Cattaneo and its neutral Spanish pack.
Now I have both Syncfusion components as Oqtane fully translated to spanish
